### ISSUE TITLE
Notepad++ hang when processing one line JSON file

### ISSUE THREAD

**Issue by nhiephon** [notepad-plus-plus/notepad-plus-plus#8747]
### Description of the Issue

Notepad++ hang when processing one line JSON file. 

### Steps to Reproduce the Issue

1. Download sample JSON file "https://drive.google.com/file/d/1MjGL47cUI-kpUTZCaOmBf4zMxxiEyMyP"
2. Open sample file with Notepad++

Size of sample file is 134 MB. However the issue still occurs with files that are smaller in size (Another sample file is 41 MB).

### Expected Behavior

Notepad++ will hang on.

### Actual Behavior

Notepad++ will hang on.

### Debug Information

Notepad++ v7.8.7   (32-bit)
Build time : Jun  5 2020 - 02:34:28
Path : C:\Program Files (x86)\Notepad++\notepad++.exe
Admin mode : OFF
Local Conf mode : OFF
OS Name : Windows 10 Enterprise (64-bit) 
OS Version : 1903
OS Build : 18362.1016
Current ANSI codepage : 1252
Plugins : DSpellCheck.dll mimeTools.dll NppConverter.dll NppExport.dll 

![image](https://user-images.githubusercontent.com/38610545/90789775-7f7ca800-e331-11ea-8ff0-e351f6a65834.png)

**Comment by nhiephon** [comment]
Any update for my report?
Expected Behavior is Notepad++ running normally. Sorry for my mistake.
Best regards,
nhiephon

**Comment by Ekopalypse** [comment]
Scintilla, the underlying control used by Npp is weak in handling really large lines. 
Npp/Scintilla isn't the right tool if you have to work with such files.

**Comment by dail8859** [comment]
@Ekopalypse I agree Scintilla is not great at handling large lines. I thought a while ago I found somewhere on a mailing list or sourceforge that Neil (the author of Scintilla) specifically states that it is a known issue handling long lines. Do you know if there is any kind of specific statement from the Scintilla project about long lines?

**Comment by Ekopalypse** [comment]
@dail8859 - I'm 100% sure that I read such a statement some time ago but wasn't able to find it, otherwise I would have linked it. And now that you are basically saying the same, I'm even more convinced that I wasn't dreaming?

**Comment by dail8859** [comment]
@Ekopalypse 

This might be what I was thinking: https://sourceforge.net/p/scintilla/bugs/1563/

> The design centre of Scintilla is user-typed source code. It is not optimized for data files with huge lines. If 10 megabyte lines are commonly used by you then you should choose a different component.

**Comment by Ekopalypse** [comment]
@dail8859 - sounds similar but not what I think it was. I thought I read something like you stated. .... known issue ... works as designed .... and that it won't get solved.
